### PR TITLE
feat: Upgrade to Binaryen 100

### PR DESCRIPTION
**Pull request by ospencer**
BREAKING CHANGE: Table name must now be provided to the `Expression.call_indirect` and `Expression.return_call_indirect` instructions.

### CODE REVIEW

**Comment by ospencer** [comment]
Tests pass locally when this is linked into the Grain compiler (with breaking changes/other issues resolved).

File: src/binaryen_stubs_expressions.c [hunk 1]
```diff
@@ -103,9 +103,11 @@ caml_binaryen_call(value _module, value _name, value _params, value _retty) {
 }
 
 CAMLprim value
-caml_binaryen_call_indirect(value _module, value _target, value _params, value _paramsty, value _retty) {
-  CAMLparam5(_module, _target, _params, _paramsty, _retty);
+caml_binaryen_call_indirect(value _module, value _table, value _target, value _params, value _paramsty, value _retty) {
+  CAMLparam5(_module, _table, _target, _params, _paramsty);
+  CAMLxparam1(_retty);
```
Comment (phated) [inline src/binaryen_stubs_expressions.c:108]: what's `xparam1`?

File: src/binaryen_stubs_expressions.c [hunk 2]
```diff
@@ -148,9 +156,13 @@ caml_binaryen_return_call_indirect(value _module, value _target, value _params,
...
   BinaryenType paramsty = BinaryenType_val(_paramsty);
   BinaryenType retty = BinaryenType_val(_retty);
-  BinaryenExpressionRef exp = BinaryenReturnCallIndirect(module, target, params, paramsLen, paramsty, retty);
+  BinaryenExpressionRef exp = BinaryenReturnCallIndirect(module, table, target, params, paramsLen, paramsty, retty);
   CAMLreturn(alloc_BinaryenExpressionRef(exp));
 }
+CAMLprim value
+caml_binaryen_return_call_indirect__bytecode(value * argv) {
+  return caml_binaryen_return_call_indirect(argv[0], argv[1], argv[2], argv[3], argv[4], argv[5]);
+}
```
Comment (phated) [inline src/binaryen_stubs_expressions.c:165]: Why did you need to add this?

**Review by phated — APPROVED** [review]
Just some questions

File: src/binaryen_stubs_expressions.c (hunk 1 above)
Comment (ospencer) [reply src/binaryen_stubs_expressions.c:108]: The macros for `CAMLparam` only go up to 5, so if you've got more you need to use the "extra" params.

File: src/binaryen_stubs_expressions.c (hunk 2 above)
Comment (ospencer) [reply src/binaryen_stubs_expressions.c:165]: If you want to call a C function from OCaml that's got more than 5 arguments, you need to provide two functions—one that does the actual work and another which calls it with the actual arguments. It's probably because OCaml's calling convention only creates functions of up to five arguments or something like that. But you'll see throughout the rest of the stubs that this is done for anything that takes more than 5.

Comment (phated) [reply src/binaryen_stubs_expressions.c:165]: Thanks for the explanation!
